### PR TITLE
Adds preference for window flashing

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -405,11 +405,11 @@ datum/projectile_data
...
 
 
-/proc/window_flash(client/C)
+/proc/window_flash(client/C, ignorepref = FALSE)
 	if(ismob(C))
 		var/mob/M = C
 		if(M.client)
 			C = M.client
-	if(!C)
+	if(!C || (!C.prefs.windowflashing && !ignorepref))
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]:
```suggestion
	if(!C?.prefs.windowflashing && !ignorepref)
```

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -405,11 +405,11 @@ datum/projectile_data
 	flick_overlay(I, viewing, duration)
 
 
-/proc/window_flash(client/C)
+/proc/window_flash(client/C, ignorepref = FALSE)
 	if(ismob(C))
 		var/mob/M = C
 		if(M.client)
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]: Given you are checking for `C` in the next line, you can remove this `if(M.client)` check.

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -405,11 +405,11 @@ datum/projectile_data
...
 			C = M.client
```
Comment (psykzz) [inline code/__HELPERS/game.dm]: Why is it that given a client it gets the mob, then resets the original client to the mobs client?

Comment (psykzz) [inline code/__HELPERS/game.dm]: Why is it that given a client it gets the mob, then resets the original client to the mobs client?

Comment (DominikPanic) [reply code/__HELPERS/game.dm]: Because unga devs can call it with a mob, just like to-chat

File: code/__HELPERS/game.dm [hunk 4]
```diff
@@ -405,11 +405,10 @@ datum/projectile_data
...
 
-/proc/window_flash(client/C)
+/proc/window_flash(client/C, ignorepref = FALSE)
 	if(ismob(C))
 		var/mob/M = C
-		if(M.client)
-			C = M.client
-	if(!C)
+		C = M.client
+	if(!C?.prefs.windowflashing && !ignorepref)
```
Comment (Rohesie) [inline code/__HELPERS/game.dm]:
```suggestion
	if(!ignorepref && !C?.prefs.windowflashing)
```
Better check if the ignore pref before. Micro optimization.

**Pull request merged**